x/stake/keeper/delegation_test.go

```diff
@@ -179,6 +179,34 @@ func TestUnbondDelegation(t *testing.T) {
 	require.Equal(t, int64(4), pool.BondedTokens)
 }
 
+// tests Get/Set/Remove/Has UnbondingDelegation
+func TestGetRedelegationsFromValidator(t *testing.T) {
+	ctx, _, keeper := CreateTestInput(t, false, 0)
+
+	rd := types.Redelegation{
+		DelegatorAddr:    addrDels[0],
+		ValidatorSrcAddr: addrVals[0],
+		ValidatorDstAddr: addrVals[1],
+		CreationHeight:   0,
+		MinTime:          0,
+		SharesSrc:        sdk.NewRat(5),
+		SharesDst:        sdk.NewRat(5),
+	}
+
+	// set and retrieve a record
+	keeper.SetRedelegation(ctx, rd)
+	resBond, found := keeper.GetRedelegation(ctx, addrDels[0], addrVals[0], addrVals[1])
+	require.True(t, found)
+
+	redelegations := keeper.GetRedelegationsFromValidator(ctx, addrVals[0])
+	require.Equal(t, 1, len(redelegations))
+	require.True(t, redelegations[0].Equal(resBond))
+
+	redelegations = keeper.GetRedelegationsFromValidator(ctx, addrVals[0])
```

Comment:
Is there some reason we're calling this twice ?

Reply:
This was the test which failed due to append affecting the state - added some more comments to make it clearer